packages/backend.ai-ui/src/components/baiClient/FileExplorer/hooks.ts

```diff
@@ -0,0 +1,57 @@
+import useConnectedBAIClient from '../../provider/BAIClientProvider/hooks/useConnectedBAIClient';
+import { VFolderFile } from '../../provider/BAIClientProvider/types';
+import { useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+
+const directoryTree: Record<string, Array<VFolderFile>> = {};
```

Comment:
Using a module-level global variable like directoryTree can indeed lead to data conflicts when multiple instances interact with it simultaneously. Moving such variables to a Context or a state management library is a better approach to ensure isolation and predictable behavior.